lib/matplotlib/projections/polar.py

```diff
@@ -282,59 +432,99 @@ def _init_axis(self):
         self._update_transScale()
 
     def _set_lim_and_transforms(self):
-        self.transAxes = BboxTransformTo(self.bbox)
+        # A view limit where the minimum radius can be locked if the user
+        # specifies an alternate origin.
+        self._originViewLim = mtransforms.LockedBbox(self.viewLim)
+
+        # Handle angular offset and direction.
+        self._direction = mtransforms.Affine2D() \
+            .scale(self._default_theta_direction, 1.0)
+        self._theta_offset = mtransforms.Affine2D() \
+            .translate(self._default_theta_offset, 0.0)
+        self.transShift = mtransforms.composite_transform_factory(
+            self._direction,
+            self._theta_offset)
+        # A view limit shifted to the correction location after accounting for
+        # orientation and offset.
+        self._shiftedViewLim = mtransforms.TransformedBbox(self.viewLim,
+                                                           self.transShift)
 
         # Transforms the x and y axis separately by a scale factor
         # It is assumed that this part will have non-linear components
-        self.transScale = TransformWrapper(IdentityTransform())
+        self.transScale = mtransforms.TransformWrapper(
+            mtransforms.IdentityTransform())
+
+        # Transforms the r-axis so that the specified origin is at r=0.
+        self.transOffset = mtransforms.ScaledTranslation(
+            0.0,
+            (self._originViewLim, 'y0'),
+            -1.0)
+
+        # The "real" view limit in consistent (theta, r) space, after applying
+        # all direction, offset, and limit options.
+        self._realViewLim = mtransforms.TransformedBbox(self._shiftedViewLim,
+                                                        self.transOffset)
+        self._realViewLim._parents[id(self)] = self
+
+        # Scale view limit into a bbox around the selected wedge. This may be
+        # smaller than the usual unit axes rectangle if not plotting the full
+        # circle.
+        self.axesLim = _WedgeBbox((0.5, 0.5), self._realViewLim)
```

Comment:
I'm not sure if `axesLim` is required to be unitized, but it isn't here, necessarily.